hack/setup-e2e-env.sh

```diff
@@ -0,0 +1,529 @@
+
+    echo -e "$help"
+    return 0
+}
+
+parse_args() {
+    while [[ $# -gt 0 ]]; do
+        case $1 in
+        -h | --help)
+            SHOW_USAGE=true
+            return 0
+            ;;
+        --validate-only)
+            VALIDATE_ONLY=true
+            shift
+            ;;
+        --no-kind)
+            INSTALL_KIND=false
+            shift
+            ;;
+        --no-kubectl)
+            INSTALL_KUBECTL=false
+            shift
+            ;;
+        --no-cluster)
+            SETUP_CLUSTER=false
+            shift
+            ;;
+        --no-olm)
+            SETUP_OLM=false
+            shift
+            ;;
+        --no-registry)
+            SETUP_REGISTRY=false
+            shift
+            ;;
+        --skip-host-check)
+            SKIP_HOST_CHECK=true
+            shift
+            ;;
+        --cluster-name)
+            shift
+            CLUSTER_NAME="$1"
+            shift
+            ;;
+        --kind-version)
+            shift
+            KIND_VERSION="$1"
+            shift
+            ;;
+        --kind-image)
+            shift
+            KIND_IMAGE="$1"
+            shift
+            ;;
+        --kubeconfig)
+            shift
+            KUBECONFIG_PATH="$1"
+            shift
+            ;;
+
+        --*)
+            die "Unknown option: $1"
+            ;;
+        *)
+            EXTRA_PACKAGES+=("$1")
+            shift
+            ;;
+        esac
+    done
+    return 0
+}
+
+check_command() {
+    local cmd="$1"
+    if command -v "$cmd" &> /dev/null; then
+        ok "$cmd is available"
+        return 0
+    else
+        warn "$cmd is not available"
+        return 1
+    fi
+}
+
+validate_prerequisites() {
+    header "Validating Prerequisites"
+
+    local fail=0
+
+    # Check for required system tools
+    check_command "go" || fail=1
+    check_command "git" || fail=1
+    check_command "curl" || fail=1
+
+    # Check for project tools that should be installed via 'make tools'
+    local tools_missing=0
+    local required_tools=("operator-sdk" "oc" "controller-gen" "kustomize")
+
+    for tool in "${required_tools[@]}"; do
+        if [[ -x "$PROJECT_ROOT_DIR/tmp/bin/$tool" ]]; then
+            ok "$tool is available in tmp/bin"
+        else
+            warn "$tool not found in tmp/bin"
+            tools_missing=1
+        fi
+    done
+
+    if [[ $tools_missing -eq 1 ]]; then
+        info "Install missing tools by running:"
+        echo "    ❯ make tools"
+        # Don't fail here as some operations might still work
+    fi
+
+    # Check /etc/hosts for local-registry entry (skip in CI)
+    if ! $SKIP_HOST_CHECK; then
+        if grep -q "local-registry" /etc/hosts 2>/dev/null; then
+            ok "/etc/hosts contains local-registry entry"
+        else
+            warn "/etc/hosts does not contain local-registry entry"
+            info "Add local-registry entry:"
+            printf "    ❯ echo \"127.0.0.1\tlocal-registry\" | sudo tee -a /etc/hosts"
```

Comment:
change as 
`printf "127.0.0.1\tlocal-registry\n" | sudo tee -a /etc/hosts`

Reply:
Maybe it'd better differentiate local_setup and github action setup, both call some common lib scripts. For me, it's difficult for me to know what should be executed locally.

Reply:
If not add the entry by executing the following, script will stop here with output https://github.com/rhobs/observability-operator/pull/791#issuecomment-3031501758
`printf "127.0.0.1\tlocal-registry\n" | sudo tee -a /etc/hosts`

If added the entry, will fail to create kind cluster with issue [COO-1048](https://issues.redhat.com/browse/COO-1048)

Reply:
`kind` also runs with docker, though not sure if that works better on a rhel system.